services/paymaster/src/constraint.rs

```diff
@@ -72,6 +72,23 @@ pub struct ContextualDomainKeys {
     pub sponsor: Pubkey,
 }
 
+fn is_compute_budget_instruction(
+    transaction: &VersionedTransaction,
+    instruction_index: usize,
+) -> bool {
+    if let Some(instruction) = transaction.message.instructions().get(instruction_index) {
+        let static_accounts = transaction.message.static_account_keys();
+        let program_id = instruction.program_id(static_accounts);
+        if program_id == &solana_compute_budget_interface::id()
+            && ComputeBudgetInstruction::try_from_slice(&instruction.data).is_ok()
```

Comment:
It bothers me a little bit to have the deserialization here, I feel like it this is not within the responsibility of this while loop which is to check that the transaction conforms to the instruction filters.

I think there's room for throwing errors if:
- there are duplicate compute budget instructions
- there are invalid compute budget instructions
But maybe they belong in the first pass inside `check_gas_spend`?